CONTRIBUTING.md

```diff
@@ -5,70 +5,106 @@ If you have any questions or suggestions we are happy to hear them.
+The completion items are being determined from the MicroProfile jdt.ls extension, which searches Quarkus project
+classpath JARs for annotations like `@ConfigRoot`.
 
- * `quarkus.application.name` for the `name` field.
- * `quarkus.application.version` for the `version` field.
- 
-In addition, the Quarkus jdt.ls extension is able to manage `Quarkus deployment JAR`. 
+Using [Eclipse Extension Points](https://www.vogella.com/tutorials/EclipseExtensionPoint/article.html),
+the MicroProfile jdt.ls extension is extensible to scan new annotations, via providers that contribute
+to the extension point.
+
+There are multiple providers that are each responsible in defining which annotation(s) to scan for.
+
+Examples:
+
+| Class | Annotations |
+|-------|-------------|
+| [MicroProfileConfigPropertyProvider](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileConfigPropertyProvider.java) | org.eclipse.microprofile.config.inject.ConfigProperty |
+| [MicroProfileRegisterRestClientProvider](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java) | org.eclipse.microprofile.rest.client.inject.RegisterRestClient |
+| [QuarkusConfigPropertiesProvider](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.jdt/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/providers/QuarkusConfigPropertiesProvider.java) | io.quarkus.arc.config.ConfigProperties |
+| [QuarkusConfigRootProvider](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.jdt/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/providers/QuarkusConfigRootProvider.java) | io.quarkus.runtime.annotations.ConfigRoot |
+| [QuarkusKubernetesProvider](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.jdt/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/providers/QuarkusKubernetesProvider.java) | io.dekorate.kubernetes.annotation.KubernetesApplication<br>io.dekorate.openshift.annotation.OpenshiftApplication<br>io.dekorate.s2i.annotation.S2iBuild<br>io.dekorate.docker.annotation.DockerBuild |
+
+
+## Searching for properties in JARs not in the user's Quarkus project's classpath:
+It is possible to scan JARs that are not in the user's Quarkus project's classpath.
+Some properties like `quarkus.application.name` are defined in runtime JARs
+and properties like `quarkus.hibernate-orm.dialect` are defined in deployment JARs.
+
+Both of these sources are not in the compile-time classpath.
```

Comment:
>It is possible to scan JARs that are not in the user's Quarkus project's classpath.
Some properties like `quarkus.application.name` are defined in runtime JARs
and properties like `quarkus.hibernate-orm.dialect` are defined in deployment JARs. Both of these sources are not in the compile-time classpath.

I'm not sure about "runtime JARs"